src/Features/CSharp/Portable/EditAndContinue/CSharpEditAndContinueAnalyzer.cs

```diff
@@ -1162,6 +1162,116 @@ internal override bool QueryClauseLambdasTypeEquivalent(SemanticModel oldModel,
             }
         }
 
+        protected override void ReportLambdaAttributeRudeEdits(SemanticModel oldModel, SyntaxNode oldLambdaBody, SemanticModel newModel, SyntaxNode newLambdaBody, List<RudeEditDiagnostic> diagnostics, bool hadSignatureEdits, CancellationToken cancellationToken)
+        {
+            // Changes from lambda to local are covered by the method above, so we can just bail out here
+            // if both old and new aren't local functions
+            if (!IsLocalFunctionBody(oldLambdaBody) || !IsLocalFunctionBody(newLambdaBody))
+            {
+                return;
+            }
+
+            var newLocalFunction = (LocalFunctionStatementSyntax)GetLambda(newLambdaBody);
+            var oldLocalFunction = (LocalFunctionStatementSyntax)GetLambda(oldLambdaBody);
+
+            var reportDiagnostic = false;
+            if (!AreEquivalentAttributeLists(oldLocalFunction.AttributeLists, newLocalFunction.AttributeLists, out var diagnosticNode))
+            {
+                reportDiagnostic = true;
+            }
+
+            // If the old and new signatures have changed then we don't need to report rude edits for attributes on parameters
+            // so we can skip this bit
+            if (!hadSignatureEdits)
+            {
+                for (var i = 0; i < oldLocalFunction.ParameterList.Parameters.Count; i++)
+                {
+                    if (!AreEquivalentAttributeLists(oldLocalFunction.ParameterList.Parameters[i].AttributeLists, newLocalFunction.ParameterList.Parameters[i].AttributeLists, out var squiggleNode))
+                    {
+                        // if the comparison didn't give us a node to highlight, then highlight the parameter
+                        diagnosticNode = squiggleNode ?? newLocalFunction.ParameterList.Parameters[i];
+                        reportDiagnostic = true;
+                        break;
+                    }
+                }
+            }
+
+            if (reportDiagnostic)
+            {
+                diagnostics.Add(new RudeEditDiagnostic(
+                    RudeEditKind.Update,
+                    GetDiagnosticSpan(diagnosticNode ?? newLocalFunction, EditKind.Update),
+                    newLocalFunction,
+                    new[] { GetDisplayName(newLocalFunction) }));
```

Comment:
>newLocalFunction [](start = 43, length = 16)

This could also take `diagnosticNode ?? newLocalFunction`, so that the error message reads "Updating 'attribute' ..." rather than "Updating 'local function' ...".